### ISSUE TITLE
Unclear error message when a nested property (source) cannot be found

### ISSUE THREAD

**Issue by sjaakd** [mapstruct/mapstruct#1354]
There's no relation to the target property in the message, which makes it hard to debug.

**Comment by filiphr** [comment]
Why should there be a relation to the target property? Wasn't the line with the nested source property marked as the error?

**Comment by sjaakd** [comment]
So to give an [example](https://github.com/BROprogramma/CPT_GEF_CONVERTER/blob/master/gef_impl/src/main/java/nl/bro/cpt/gef/transform/GefToBroMapperImbro.java)

```java
    @Mapping( target = "requestReference", source = "requestReference" )
    @Mapping( target = "deliveryAccountableParty", source = " survey.gefCptFile.measurementText101" )
    @Mapping( target = "qualityRegime", constant = "IMBRO" )
    @Mapping( target = "sourceDocument.CPT", source = "survey" )
    @Mapping( target = "broId", ignore = true )
    @Mapping( target = "underPrivilege", ignore = true )
    public abstract RegistrationRequestType gefToRegistrationRequest(GefCptSurvey survey, String requestReference);
```

will give an error message `nl/bro/cpt/gef/transform/GefToBroMapperImbro.java:[96,45] Method has no parameter named " survey".`. Note that in case of nested mappings, MapStruct refers to a parameter (and hence the method). However, it would be better understandable if we would refer to the line that contains the wrong`@Mapping` or even to the `target`

**Comment by filiphr** [comment]
Now I get it, thanks. Yes we should highlight the line with the `@Mapping` annotation and the `source` value, as that one is the one which is wrong.

**Comment by sjaakd** [comment]
The issue has been resolved by trimming (#1353). The problem was caused by the leading space in `source = " survey.gefCptFile.measurementText101"`. That causes MapStruct to associate the error message with the parameter (`" survey"`) which is not there and hence gives an error on the method instead of the annotation.

Trimming (#1354) solves this and causes MapStruct to give the error on the proper location (the `@Mapping` annotation). It also states clearly parameter and property now.

So: problem fixed.
